client/handler.go

```diff
@@ -307,6 +441,64 @@ func (h *Handler) Get(w http.ResponseWriter, r *http.Request, ps httprouter.Para
 	h.r.Writer().Write(w, r, c)
 }
 
+// swagger:route GET /dyn-clients/{id} public getDynOAuth2Client
+//
+// Get an OAuth 2.0 Client.
+//
+// Get an OAUth 2.0 client by its ID. This endpoint never returns passwords.
+//
+// OAuth 2.0 clients are used to perform OAuth 2.0 and OpenID Connect flows. Usually, OAuth 2.0 clients are generated for applications which want to consume your OAuth 2.0 or OpenID Connect capabilities. To manage ORY Hydra, you will need an OAuth 2.0 Client as well. Make sure that this endpoint is well protected and only callable by first-party components.
+//
+//
+//     Consumes:
+//     - application/json
+//
+//     Produces:
+//     - application/json
+//
+//     Schemes: http, https
+//
+//     Responses:
+//       200: oAuth2Client
+//       401: genericError
+//       500: genericError
+func (h *Handler) GetDynamicRegistration(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	var id = ps.ByName("id")
+	var secret = r.URL.Query()["secret"][0]
+
+	c, err := h.r.ClientManager().Authenticate(r.Context(), id, []byte(secret))
+	if err != nil {
+		err = herodot.ErrUnauthorized.WithReason("The requested OAuth 2.0 client does not exist or you did not provide the necessary credentials")
+		h.r.Writer().WriteError(w, r, err)
+		return
+	}
+
+	c.Secret = ""
+	if c.Metadata != nil {
```

Comment:
Metadata has a different meaning. It's purpose is not to store configuration for the client, but to store any arbitrary data. Adding `access_token_ttl` here is not metadata, imo, but configuration. Therefore I think that it would be better to have, for example, on the admin port:

```
POST /clients/<id>/settings
```

which would allow to change the config of clients - so for example the access token TTL.

Reply:
Repurposing this field for some other functionality makes Ory Hydra vulnerable. Say you have Hydra 1.10 running. Someone scans you and sees that you use Ory Hydra 1.10 and have this endpoint exposed. He/she adds config options to the client in the metadata field. One day, you upgrade to Ory Hydra 1.11. Now, the attacker was able to change configuration which he was not supposed to change.

Therefore, this has to go somewhere else! It's generally a good idea to never use untrusted data as trusted data. Validation might help but it often fails to cover all areas :)

Reply:
I understand.....  How about have and `access_token_ttl` or `access_token_ttl` as specific fields in the root level? As now we will different endpoints for Dynamic Client registration and the `/client `ones in the admin API. Then we can have different client representations and we can include those `access_token_ttl` or `access_token_ttl` only in the one for admin API... Then hydra master can keep that configuration in sync... Does it make sense?

Reply:
Javier, I'm somehow lost by your question:
> Separate the client representation for Dynamic Client registration and the admin api maybe can be done in separate PR?

Reply:
@dadrus I mean to have `access_token_ttl` or `access_token_ttl` as dedicated fields fields for the client entity and not as part of metadata...

For this PR we will separate the public and admin endpoints... making public ones for Dynamic Client Registration to do no accept any metadata.

Does it make sense?